pkg/storage/mock/storer.go

```diff
@@ -6,28 +6,41 @@ package mock
 
 import (
 	"context"
+	"sync"
 
 	"github.com/ethersphere/bee/pkg/storage"
 	"github.com/ethersphere/bee/pkg/swarm"
 )
 
+type MockStorer interface {
+	storage.Storer
+	GetModeSet(addr swarm.Address) (mode storage.ModeSet)
+}
+
 type mockStorer struct {
 	store     map[string][]byte
+	modeSet   map[string]storage.ModeSet
+	modeSetMu *sync.Mutex
```

Comment:
mockStorer type is used with pointer semantics, so this mutex can be a value field.